client/signup/steps/mlb-domains/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+var React = require( 'react' );
+
+/**
+ * Internal dependencies
+ */
+var StepWrapper = require( 'signup/step-wrapper' ),
+	Input = require( 'components/forms/form-text-input-with-affixes' ),
+	SelectDropdown = require( 'components/select-dropdown' ),
+	productsList = require( 'lib/products-list' )(),
+	Card = require( 'components/card' ),
+	SimpleNotice = require( 'notices/simple-notice' ),
+	SignupActions = require( 'lib/signup/actions' );
+
+module.exports = React.createClass( {
+	displayName: 'MlbDomainsStep',
+
+	getInitialState: function() {
+		return { blogName: '' };
+	},
+
+	handleNameChange: function( event ) {
+		event.preventDefault();
+		this.setState( { blogName: event.target.value } );
+	},
+
+	handleSubmit: function( event ) {
+		event.preventDefault();
+
+		const siteUrl = this.state.blogName,
+			domainItem = undefined;
+
+		SignupActions.submitSignupStep( {
+			processingMessage: this.translate( 'Adding your domain' ),
+			stepName: this.props.stepName,
+			siteUrl,
+			stepSectionName: this.props.stepSectionName
+		}, [], { domainItem } );
+
+		this.props.goToNextStep();
+	},
+
+	renderForm: function() {
+		return (
+			<Card>
+				<label htmlFor="domain">Blog Address</label>
+				<Input placeholder="yourname" onChange={ this.handleNameChange } name="domain" suffix={ this.renderProducts() }/>
+				<button onClick={ this.handleSubmit } className='button is-primary'>
+					{ this.translate( 'Continue' ) }
+				</button>
+
+			</Card>
+			);
+	},
+
+	renderProducts: function() {
+		var options = [],
+			products = productsList.get();
+		products = Object.keys( products ).map( ( k ) => products[ k ] );
+		options = [ { value: '.mlblogs.com', label: '.mlbslogs.com Free' } ];
+
+		products = products.filter( p => p.is_domain_registration );
+		// Filter domaing registration. (Not present on mlblogs.com)
```

Comment:
Is it OK if I replace all of this, with the site Step? (leaving those domains offered on current signup flow).

I''l need to alter [this line](https://github.com/Automattic/wp-calypso/blob/master/client/signup/steps/site-creation/index.jsx#L246) on the `site`step and add a `getDefaultProps()` for showing `.wordpress.com` by default and being able to reuse the component.